### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by ChriPiv**
Changes:
-changed doubles/floats to number
-added missing parameters to BarcodePicker.show() function
-added missing boolean types to several functions
-added explanation how searchbar result callback works
-added explanation how the units of setMargin work
-added documentation how to build the sample apps
-added a new "simple" sample app (first example from the README.md file)

### CODE REVIEW

**Comment by ChriPiv** [comment]
I'm not sure if changing double/floats to number is the right thing - 
On one hand I feel its wrong to ask for "doubles" or "float", since they don't exist in javascript
On the other hand it's not really consistent, since we keep the "int"s
What do you think?

File: samples/simple/www/index.html [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <meta name="format-detection" content="telephone=no" />
+        <meta name="msapplication-tap-highlight" content="no" />
+        <!-- WARNING: for iOS 7, remove the width=device-width and height=device-height attributes. See https://issues.apache.org/jira/browse/CB-4323 -->
+        <meta name="viewport" content="user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width, height=device-height, target-densitydpi=device-dpi" />
+        <link rel="stylesheet" type="text/css" href="css/index.css" />
+        <title>Scandit Barcode Scanner</title>
+    </head>
+    <body onload="onBodyLoad()" style="background: url(img/ScanditSDKDemo-Splash.png) no-repeat;background-size: 100%;background-color: #000000">
+        <script type="text/javascript" src="cordova.js"></script>
+        <script type="text/javascript" src="js/index.js"></script>
```
Comment (biasmv) [inline samples/simple/www/index.html]: I don't think js/index.js exists. It's also not required in the sample. Can you remove this line?

File: docs/Scandit.java [hunk 2]
```diff
@@ -304,8 +304,15 @@
          * fully configured. Once it is shown make sure to call
          * {@link startScanning() startScanning()} if you want to start the video feed and scan for
          * barcodes.
+         * The success callback takes a {@link ScanSession} as parameter containing the information
+         * about the scanned codes. The manual callback takes a String as parameter containing the
+         * data.
+         *
+         * @param success callback for successfull scan
+         * @param manual callback for manual scan using search bar
+         * @param failure callback if operation fails
```
Comment (biasmv) [inline docs/Scandit.java]: I'd change this to:

``
@param success callback to be invoked whenever codes have been successfully scanned. The callback receives the scan session as the first and only argument which contains a list of recognized codes.
@param manual callback to be invoked when the user enters a text in the search bar. The entered text is passed as the first argument to the callback. If you do not use the search bar, you may pass null.
@param failure callback to be invoked upon failure, or when cancel is called on the picker. The callback is passed a reason for failure.
``

File: docs/Scandit.java [hunk 3]
```diff
@@ -958,7 +967,10 @@ public void setViewfinderDimension(
 		///@{
 		
 		/**
-		 * Shows (or hides) a search bar at the top of the scan screen.
+		 * Shows (or hides) a search bar at the top of the scan screen. When data is
+		 * submitted from the search bar the callback specified in
+		 * {@link BarcodePicker.show BarcodePicker.show} is calledfloat passed as
+		 * parameter in a string.
```
Comment (biasmv) [inline docs/Scandit.java]: The last sentence doesn't seem right

File: docs/cropped-or-scaled-picker.md [hunk 4]
```diff
@@ -17,13 +17,19 @@ The plugin allows you to display it as a cropped and/or scaled subview such that
...
+~~~~~~~~~~~~~~~~
+
+It is possible to adjust these margins while the scanner is running by using the resize function. The last argument is the animation duration which you only want to set to something greater than 0 if the picker is already displayed and should animate to a new size:
 
 ~~~~~~~~~~~~~~~~{.java}
 picker.setMargins(new Scandit.Margins(0, 0, 0, 100), new Scandit.Margins(0, 0, 100, 0), 0.5);
 ~~~~~~~~~~~~~~~~
 
-This code would reduce the margins to 100 points and animate that change over 0.5 seconds.
+This code would reduce the margins to 100 units and animate that change over 0.5 seconds.
```
Comment (biasmv) [inline docs/cropped-or-scaled-picker.md]: I'd write this as "This code will reduce the margins..."

**Comment by ChriPiv** [comment]
This should fix all mentioned issues

**Pull request closed without merging**